src/TeamSyncer.ts

```diff
@@ -0,0 +1,317 @@
+                matrix_room_id: roomId,
+                slack_channel_id: channelItem.id,
+                team_id: teamId,
+            });
+        } catch (ex) {
+            log.error("Failed to provision new room dynamically:", ex);
+        }
+    }
+
+    public async onChannelDeleted(teamId: string, channelId: string) {
+        log.info(`${teamId} removed channel ${channelId}`);
+        if (!this.shouldTeamSync(teamId, "channel", channelId)) {
+            return;
+        }
+        const room = this.main.rooms.getBySlackChannelId(channelId);
+        if (!room) {
+            log.warn("Not unlinking channel, no room found");
+            return;
+        }
+
+        try {
+            await this.main.botIntent.sendMessage(room.MatrixRoomId, {
+                msgtype: "m.notice",
+                body: "The Slack channel bridged to this room has been deleted.",
+            });
+        } catch (ex) {
+            log.warn("Failed to send deletion notice into the room:", ex);
+        }
+
+        // Hide deleted channels in the room directory.
+        try {
+            await this.main.botIntent.getClient().setRoomDirectoryVisibility(room.MatrixRoomId, "private");
+        } catch (ex) {
+            log.warn("Failed to hide room from the room directory:", ex);
+        }
+
+        try {
+            await this.main.actionUnlink({ matrix_room_id: room.MatrixRoomId });
+        } catch (ex) {
+            log.warn("Tried to unlink room but failed:", ex);
+        }
+    }
+
+    public async onTeamMemberJoined(teamId: string, user: any) {
+        log.info(`${teamId} added user ${user}`);
+    }
+
+    public async onUserChange(teamId: string, user: any) {
+        log.info(`${teamId} added user ${user}`);
+    }
+
+    private getAliasPrefix(teamId: string) {
+        const channelConfig = this.teamConfigs[teamId].channels;
+        if (!channelConfig) {
+            return;
+        }
+        return channelConfig.alias_prefix;
+    }
+
+    private async createRoomForChannel(teamId: string, creator: string, channel: ConversationsInfo): Promise<string> {
+        let intent;
+        let creatorUserId: string|undefined;
+        try {
+            creatorUserId = (await this.main.getGhostForSlack(creator, undefined, teamId)).userId;
+            intent = this.main.getIntent(creatorUserId);
+        } catch (ex) {
+            // Couldn't get the creator's mxid, using the bot.
+            intent = this.main.botIntent;
+        }
+        const aliasPrefix = this.getAliasPrefix(teamId);
+        const alias = aliasPrefix ? `${aliasPrefix}${channel.name.toLowerCase()}` : undefined;
+        let topic: undefined|string;
+        if (channel.purpose) {
+            topic = channel.purpose.value;
+        }
+        log.debug("Creating new room for channel", channel.name, topic, alias);
+        const plUsers = {};
+        plUsers[this.main.botUserId] = 100;
+        if (creatorUserId) {
+            plUsers[creatorUserId] = 100;
```

Comment:
can we have the ability to add users to this list via config? I will want to add matrix side admins to all the rooms etc.